cli-plugins/socket/socket_test.go

```diff
@@ -25,12 +24,11 @@ func TestSetupConn(t *testing.T) {
 		_, err = net.DialUnix("unix", nil, addr)
 		assert.NilError(t, err, "failed to dial returned listener")
 
-		pollConnNotNil(t, &conn)
+		pollConnNotNil(t, conn)
 	})
 
 	t.Run("allows reconnects", func(t *testing.T) {
```

Comment:
So supposedly, this test should ensure that we let plugins redial the connection, but in reality the test is a little incomplete – we're ensuring that redialing the connection doesn't error out, but not that we can use it.

Reply:
For clarity, imagine a test such as:
```go
listener, conn, err := SetupConn()
assert.NilError(t, err)
assert.Check(t, listener != nil, "returned nil listener but no error")
addr, err := net.ResolveUnixAddr("unix", listener.Addr().String())
assert.NilError(t, err, "failed to resolve listener address")

otherConn, err := net.DialUnix("unix", nil, addr)
assert.NilError(t, err, "failed to dial returned listener")
pollConnNotNil(t, conn)

otherConn.Close()

_, err = net.DialUnix("unix", nil, addr)

pollConnNotNil(t, conn)
assert.NilError(t, err, "failed to redial listener")
```
(this fails)